src/main/java/org/example/learningsystem/core/template/service/MustacheTemplateService.java

```diff
@@ -0,0 +1,37 @@
+package org.example.learningsystem.core.template.service;
+
+import com.samskivert.mustache.Mustache;
+import lombok.extern.slf4j.Slf4j;
+import org.example.learningsystem.core.template.exception.RenderTemplateException;
+import org.springframework.boot.autoconfigure.mustache.MustacheResourceTemplateLoader;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class MustacheTemplateService implements RenderTemplateService {
+
+    private static final String TEMPLATES_PREFIX = "templates/";
+    private static final String TEMPLATES_SUFFIX = ".mustache";
+
+    private final MustacheResourceTemplateLoader templateLoader =
+            new MustacheResourceTemplateLoader(TEMPLATES_PREFIX, TEMPLATES_SUFFIX);
+
+    @Override
+    public String render(String path, Object arguments) {
+        try {
+            return tryRender(path, arguments);
+        }
+        catch (Exception e) {
+            log.error("Error during rendering {}: {}", path, e.getMessage());
```

Comment:
you can pass this message into exception without writing log here, because anyway Exception Handler will catch this exception and will write messages from exceptions stack trace